sympy/functions/special/delta_functions.py

```diff
@@ -47,6 +81,36 @@ class DiracDelta(Function):
     is_real = True
 
     def fdiff(self, argindex=1):
+        """
+        Returns the first derivative of a DiracDelta Function.
+
+        The difference between ``diff()`` and ``fdiff()`` is:-
+        ``diff()`` is the user-level function and ``fdiff()`` is an object method.
+        ``fdiff()`` is a convenience. It lets the user write the derivative without
+        concern for the chain rule.
+        ``diff(expr, x)`` calls ``expr._eval_derivative(x)`` internally,
+        if ``expr`` is a subclass of ``Function``, then the ``Function._eval_derivative``
+        calls ``expr.fdiff``.
```

Comment:
``fdiff`` is just a convenience method available in the ``Function`` class. It returns the derivative of the function without considering the chain rule.``diff(function, x)`` calls  ``Function._eval_derivative``which in turn calls ``fdiff`` internally to compute the derivative of the function.